### PR TITLE
Session management

### PR DESCRIPTION
**Pull request by decanus**

### CODE REVIEW

File: status-session-management-spec.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
+# Status Session Management Specification
+
+> Version: 0.1 (Draft)
+>
+> Authors: Andrea Piana <andreap@status.im>, Pedro Pombeiro <pedro@status.im>, Corey Petty <corey@status.im>, Oskar Thorén <oskar@status.im>, Dean Eigenmann <dean@status.im>
```
Comment (oskarth) [inline status-session-management-spec.md]: alphabetical order

File: status-session-management-spec.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+## Abstract
+
+In this specification we describe how status sessions are handled.
```
Comment (oskarth) [inline status-session-management-spec.md:9]: what's a session?

File: status-session-management-spec.md [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+  - [Account recovery](#account-recovery)
+  - [Partitioned devices](#partitioned-devices)
+  - [Trust establishment](#trust-establishment)
+    - [Contact request](#contact-request)
+  - [Expired session](#expired-session)
+  - [Stale devices](#stale-devices)
+
+## Introduction
+
+A peer is identified by two pieces of data:
```
Comment (oskarth) [inline status-session-management-spec.md:31]: how does this connect to a session in lifecycle

File: status-session-management-spec.md [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+This mean that every time a new device is paired, the bundle needs to be updated and propagated with the new information, and the burden is put on the user to make sure the pairing is successful.
+
+The method is loosely based on https://signal.org/docs/specifications/sesame/ .
+
+<!-- TODO: This multi device section isn't clear enough -->
+<!-- TODO: Additionally, it seems tightly coupled with secure transport, which makes things like multi device public chats harder to reason about (IMO). E.g. as a client impl I might want multi device support but not want to impl double ratchet etc, so what does this mean? -->
+
+## Pairing
+
+When a user adds a new account in the `Status` application, a new `installation-id` will be generated. The device should be paired as soon as possible if other devices are present. Once paired the contacts will be notified of the new device and it will be included in further communications.
```
Comment (oskarth) [inline status-session-management-spec.md:65]: Is this all part of session?

File: status-session-management-spec.md [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+## Account recovery
+
+Account recovery is no different from adding a new device, and it is handled in exactly the same way.
+
+## Partitioned devices
+
+In some cases (i.e. account recovery when no other pairing device is available, device not paired), it is possible that a device will receive a message that is not targeted to its own `installation-id`.
+In this case an empty message containing bundle information is sent back, which will notify the receiving end of including this device in any further communication.
+
+## Trust establishment
```
Comment (oskarth) [inline status-session-management-spec.md:90]: does this belong in session management?

File: status-session-management-spec.md (hunk 4 above)
Comment (decanus) [reply status-session-management-spec.md:65]: Probably not, but I don't see where else we would add this pairing info. Accounting?

**Comment by oskarth** [comment]
Merging for now to share progress, comments above still applies

**Comment by oskarth** [comment]
@decanus can you add a link to session management from other spec and add to readme? in addition to above comments

**Pull request merged**